### PR TITLE
Fix buffer overflow that happens when processing the very first sample

### PR DESCRIPTION
**Pull request by johnnovak**

### CODE REVIEW

File: src/libs/tal-chorus/Chorus.h [hunk 1]
```diff
@@ -108,6 +105,8 @@ class Chorus
 			*writePtr= 0.0f;
 		}
 		while (++writePtr < delayLineEnd);
+
+		writePtr = delayLineStart;
```
Comment (johnnovak) [inline src/libs/tal-chorus/Chorus.h:109]: Actual fix ☝🏻

Comment (kcgen) [reply src/libs/tal-chorus/Chorus.h:109]: If we re-assign `writePtr = delayLineStart` after the prior while-loop incrementing, then the final incremented `writePtr` value is overwritten.

Seems like we can drop the while-loop?

**Comment by kcgen** [comment]
Nice bounds-checks after each length increment! :+1:

**Review by kcgen — APPROVED** [review]
Amazing quick fix, @johnnovak !

(approving; just one comment for fixup; feel free to merge when ready :+1:)

File: src/libs/tal-chorus/Chorus.h (hunk 1 above)
Comment (johnnovak) [reply src/libs/tal-chorus/Chorus.h:109]: Hmm, we need it for zeroing out the buffer.

Comment (kcgen) [reply src/libs/tal-chorus/Chorus.h:109]: Ahh sorry - the "do { .. }" body was collapsed in GitHub :sweat_smile:  (All I was seeing was an empty while loop; makes sense now!)

**Pull request merged**